atomicapp/providers/lib/kubebase.py

```diff
@@ -0,0 +1,309 @@
+"""
+ Copyright 2014-2016 Red Hat, Inc.
+
+ This file is part of Atomic App.
+
+ Atomic App is free software: you can redistribute it and/or modify
+ it under the terms of the GNU Lesser General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ Atomic App is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU Lesser General Public License for more details.
+
+ You should have received a copy of the GNU Lesser General Public License
+ along with Atomic App. If not, see <http://www.gnu.org/licenses/>.
+"""
+
+import requests
+import websocket
+import tempfile
+import base64
+import ssl
+from requests.exceptions import SSLError
+from atomicapp.providers.lib.exceptions import (KubeBaseError,
+                                                KubeConnectionError)
+from atomicapp.constants import LOGGER_DEFAULT
+import logging
+logger = logging.getLogger(LOGGER_DEFAULT)
+
+
+class KubeBase(object):
+
+    '''
+    The role of Kube Base is to parse the Kube Config file and create an
+    understandable API as well as initiation of connection to
+    Kubernetes-based APIs (OpenShift/Kubernetes).
+
+    '''
+    cluster = None
+    user = None
+    token = None
+
+    def __init__(self, config):
+        '''
+        Args:
+            config (object): An object of the .kube/config configuration
+        '''
+        self.config = config
+
+        # Gather the "current-context" from .kube/config which lists what the
+        # associated cluster, user, token, etc. is being used.
+        if "current-context" not in config:
+            raise KubeBaseError("'current-context' needs to be set within .kube/config")
+        else:
+            self.current_context = config["current-context"]
+
+        # Gather the user, context and cluster details of .kube/config based upon the current_context
+        kubeconfig_context = self._contexts()[self.current_context]
+        kubeconfig_user = kubeconfig_context['user']
```

Comment:
ACK. I added 'user' due to picking up the "token" information. You are correct that it is not needed. I will update the PR for this edge-case.